OSXvnc-server/main.c

```diff
@@ -563,13 +570,35 @@ char *rfbGetFramebuffer(void) {
     if (floor(NSAppKitVersionNumber) > floor(NSAppKitVersionNumber10_6)) {
         if (!frameBufferData) {
             CGDirectDisplayID mainDisplayID = CGMainDisplayID();
-            CGImageRef imageRef = CGDisplayCreateImage(mainDisplayID);
+            CGImageRef imageRef;
+            //Check to see if Retina Display
+            if (floor(NSAppKitVersionNumber) > floor(NSAppKitVersionNumber10_6) & displayScale > 1) {
```

Comment:
Do we need the check of the version number? If yes, replace & by &&.
As far as I know, retina displays started with 10.7, so displayScale > 1 implies version > 10.6.